Wikipedia/UI-V5/WMFArticleListCollectionViewController.m

```diff
@@ -126,10 +128,34 @@ - (NSString*)debugDescription {
     return [NSString stringWithFormat:@"%@ dataSourceClass: %@", self, [self.dataSource class]];
 }
 
-- (void)refreshVisibleCells {
-    [self.collectionView wmf_enumerateVisibleCellsUsingBlock:
-     ^(WMFArticlePreviewCell* cell, NSIndexPath* path, BOOL* _) {
+#pragma mark - Stay Fresh... yo
+
+- (void)observeArticleUpdates {
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:MWKArticleSavedNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(articleUpdatedWithNotification:) name:MWKArticleSavedNotification object:nil];
+}
+
+- (void)unobserveArticleUpdates {
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:MWKArticleSavedNotification object:nil];
+}
+
+- (void)articleUpdatedWithNotification:(NSNotification*)note {
+    MWKArticle* article = note.userInfo[MWKArticleKey];
+    [self refreshAnyVisibleCellsWhichAreShowingTitle:article.title];
+}
+
+- (void)refreshAnyVisibleCellsWhichAreShowingTitle:(MWKTitle*)title {
+    NSArray* indexPathsToRefresh = [[self.collectionView indexPathsForVisibleItems] bk_select:^BOOL (NSIndexPath* path) {
+        WMFArticlePreviewCell* cell = (WMFArticlePreviewCell*)[self.collectionView cellForItemAtIndexPath:path];
+        return (cell.title == title);
     }];
+    if (indexPathsToRefresh.count > 0) {
+        [self.collectionView reloadItemsAtIndexPaths:indexPathsToRefresh];
+    }
+}
+
+- (void)dealloc {
+    [self unobserveArticleUpdates];
```

Comment:
If you do DidLoad only, which is fine, make sure to remove I unobserve from
didDisappear
On Fri, Oct 16, 2015 at 12:44 Brian Gerstle <brian.gerstle@gmail.com> wrote:

> Did you observe _and_ update in willAppear? Should work
> On Fri, Oct 16, 2015 at 12:43 montehurd <notifications@github.com> wrote:
>
>> In Wikipedia/UI-V5/WMFArticleListCollectionViewController.m
>> <https://github.com/wikimedia/wikipedia-ios/pull/176#discussion_r42263040>
>> :
>>
>> >      }];
>> > +    if (indexPathsToRefresh.count > 0) {
>> > +        [self.collectionView reloadItemsAtIndexPaths:indexPathsToRefresh];
>> > +    }
>> > +}
>> > +
>> > +- (void)dealloc {
>> > +    [self unobserveArticleUpdates];
>>
>> Ended up moving the start observation method to viewDidLoad, so ended up
>> needing this after all.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/wikimedia/wikipedia-ios/pull/176/files#r42263040>.
>>
>